src/scheduler/fifo.c

```diff
@@ -2750,12 +2731,49 @@ update_svr_schedobj(int connector, int cmd, int alarm_time)
 	}
 	patt->next = NULL;
 
-	err = pbs_manager(connector,
-			  MGR_CMD_SET, MGR_OBJ_SCHED,
-			  sc_name, attribs, NULL);
-	if (err == 0)
-		svr_knows_me = 1;
+	pbs_manager(connector, MGR_CMD_SET, MGR_OBJ_SCHED, sc_name, attribs, NULL);
 
 	free(attribs);
 	return 1;
 }
+
+/**
+ * @brief
+ *	Validates the sched object attributes changed from Server.
+ *
+ * @param[in] connector - socket descriptor to server
+ *
+ * @retval Error code
+ * @return 0 - Failure
+ * @return 1 - Success
+ *
+ * @par Side Effects:
+ *	None
+ *
+ *
+ */
+int
+validate_sched_attrs(int connector)
+{
+	struct batch_status *ss = NULL;
+	struct batch_status *all_ss = NULL;
+
+	/* Stat the scheduler to get details of sched */
+
+	all_ss = pbs_statsched(connector, NULL, NULL);
+	ss = bs_find(all_ss, sc_name);
+
+	if (ss == NULL) {
+		pbs_statfree(all_ss);
```

Comment:
The idea is not to log anything inside the function validate_sched_attrs. Instead we can log wherever it is called. So I have added the log message inside update_svr_schedobj() function. 
I thought it is not good to add the same log message i.e. "Unable to retrieve the scheduler attributes from server" because validate_sched_attrs() is where we refactored the code and it is now doing more than fetching scheduler stats.

Reply:
Ok, but ss == NULL here actually does imply "Unable to retrieve the scheduler attributes from server" right? A generic message saying "validate_sched_attrs failed" might not be as useful. One option is to do "validate_sched_attrs: Unable to retrieve the scheduler attributes from server"

Reply:
@suresh-thelkar the scheduler generally logs where the error happens itself and not in the caller.  This way we can get a more specific reason of what went wrong.

Reply:
@bhroam / @agrawalravi90, I have now moved the log message inside the function validate_sched_attrs itself. Please provide further review comments if any or sign off.